refactron/cicd/github_actions.py

```
@@ -0,0 +1,247 @@
+
+from pathlib import Path
+from typing import Dict, Optional
+
+
+class GitHubActionsGenerator:
+    """Generate GitHub Actions workflow templates for Refactron."""
+
+    @staticmethod
+    def generate_analysis_workflow(
+        python_versions: Optional[list[str]] = None,
+        trigger_on: list[str] = None,
```

Ruff (0.14.13)

PEP 484 prohibits implicit `Optional`

Convert to `Optional[T]`

(RUF013)